### ISSUE TITLE
`WINIT_HIDPI_FACTOR` does not work when `Xft.dpi` is set

### ISSUE THREAD

**Issue by Toqozz** [rust-windowing/winit#1141]
This is a bit of a weird one.  I **think** it's tied to `Xft.dpi` but I'm not positive.

On X11, the `WINIT_HIDPI_FACTOR` environment variable seems to be ignored if `Xft.dpi` is set in the user's `.Xresources` file:

Xresources:
```
Xft.dpi: 80
```
Program:
```rust
std::env::set_var("WINIT_HIDPI_FACTOR", "1.0");

let event_loop = EventLoop::new();

let window = WindowBuilder::new()
    .with_title("A fantastic window!")
    .build(&event_loop)
    .unwrap()

dbg!(window.hidpi_factor());
```
Output:
```
[src/main.rs:20] window.hidpi_factor() = 0.8333333333333334
```

This behaviour continues even if you remove `Xft.dpi: 80` from the `.Xresources` file.  The only way to get winit to respect `WINIT_HIDPI_FACTOR` again is to set an invalid value like `Xft.dpi: foo` -- perhaps the value is being cached somewhere?

My guess is that the hidpi factor gets overwritten by the Xresources config regardless of `WINIT_HIDPI_FACTOR` at some point.

`winit = "0.20.0-alpha3"`

**Comment by Toqozz** [comment]
Indeed;
https://github.com/rust-windowing/winit/blob/5bc3cf18d98aef0a5f0a51c0cf655d37af833730/src/platform_impl/linux/x11/util/randr.rs#L109-L119

`WINIT_HIDPI_FACTOR` is accounted for in `calc_dpi_factor()`, but doesn't run if an xft dpi exists.

I'd be interested in contributing a fix for this.

We probably want to move checking for `WINIT_HIDPI_FACTOR` out of `calc_dpi_factor()`.  Maybe something like `get_dpi()` which first checks for `WINIT_HIDPI_FACTOR` and then proceeds as it does currently?

**Comment by Osspial** [comment]
FWIW, I don't think this will be relevant once the #939 work is complete, since we're largely stopping doing the HiDPI factor transformation internally.

**Comment by Toqozz** [comment]
Great!

**Comment by rofrol** [comment]
If https://github.com/rust-windowing/winit/issues/939 is close, can this also be closed?

**Comment by murarth** [comment]
Yes. Closed by #1354.
